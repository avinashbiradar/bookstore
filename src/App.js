
import './App.css';
import {BrowserRouter,Switch,Route} from 'react-router-dom'
import Login from "./Components/LogIn/LogIn"
import Dashboard from "./Components/dashBoard/dashBoard"
import Cart from "./Components/cart/cart"
import AuthRoute from './Components/authroute/authroute';
import AuthAdminRoute from './Components/authroute/authadminroute';
import PrivateRoute from './Components/privateroute/privateroute';
import PrivateAdminRoute from './Components/privateroute/privateadminroute';
import AdminBooks from './Components/admin/getadminbooks'
import AdminLogin from './Components/admin/adminlogin'
import AdminSignUp from "./Components/admin/adminRegister"
import wishlist from "./Components/wishlist/wishlist"
import NotFound from "./Components/ErrorPage/error"
function App() {

  return (
    <div className="App">
    <BrowserRouter >
    <Switch>
     <PrivateRoute path="/dashboard" component={Dashboard} />
     <AuthRoute  path="/Login" component={Login} exact/>
     <PrivateRoute path="/cart" component={Cart} exact/>
     <AuthAdminRoute path="/loginadmin" component={AdminLogin} exact/>
     <PrivateAdminRoute path="/adminbooks" component={AdminBooks} exact/>
     <AuthAdminRoute path="/adminsignup" component={AdminSignUp} exact/>
     <Route component={NotFound} />
   </Switch>
   </BrowserRouter >
  </div>
  );
}

export default App;
