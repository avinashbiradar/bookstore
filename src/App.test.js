import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react'
import App from './App';
afterEach(cleanup)
 
 it('should take a snapshot', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
   })


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App/>, div);
  ReactDOM.unmountComponentAtNode(div);
});