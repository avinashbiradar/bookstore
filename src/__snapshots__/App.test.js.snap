// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="image-content"
      >
        <h3>
          404 - Page Not Found 
        </h3>
        <p>
          This isn't the page you're looking for .
        </p>
        <p>
          Please click on the below link "Go Home..." to continue your surfing 
        </p>
        <p />
      </div>
      <div
        class="link-content"
      >
        <a
          href="/login"
        >
          <h1>
            Go to login . . .
          </h1>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
